R/stat-density.r

```diff
@@ -15,6 +15,8 @@
 #'   not line-up, and hence you won't be able to stack density values.
 #'   This parameter only matters if you are displaying multiple densities in
 #'   one plot or if you are manually adjusting the scale limits.
+#' @param bounds Known lower and upper bounds for estimated data. Default
+#'   `c(-Inf, Inf)` means that there are no (finite) bounds.
```

Comment:
There should be some explanation somewhere in the documentation of what happens when the bounds are finite. And what happens if data points fall outside the set bounds?

Reply:
This was more of an open question to you. What do you think is the best behavior here? It mostly depends on the implied severity of `bounds` argument.

Currently no check is done. It estimates density and then tries to "reflect" it at edges. There are some extreme cases (like when bounds and data range not intersect at all), which I decided to tackle after the decision about `bounds` severity.

I can suggest at least three options:
- Give a warning when at least one point is outside of `bounds`. Proceed as nothing happened. Implies that `bounds` is essentially for visualization purpose.
- Remove points outside of `bounds` before or after estimating density. This implies that `bounds` is a data generation assumption and tries to fix it.
- Give an error if at least one point is outside. This also implies data generation but stops early. I personally like this the most.

Reply:
This was more of an open question to package authors. What do you think is the best behavior here? It mostly depends on the implied severity of bounds argument.

Currently no check is done. It estimates density and then tries to "reflect" it at edges. There are some extreme cases (like when bounds and data range not intersect at all), which I decided to tackle after the decision about bounds severity.

I can suggest at least three options:
- Give a warning when at least one point is outside of bounds. Proceed as nothing happened. Implies that bounds is essentially for visualization purpose.
- Remove points outside of bounds before or after estimating density. This implies that bounds is a data generation assumption and tries to fix it.
- Give an error if at least one point is outside. This also implies data generation but stops early. I personally like this the most.